src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,284 @@
+    STOP(Degrees.of(0)), // Stop the arm
+    STOWED(Degrees.of(0)), // Arm tucked in
+    INTAKE(Degrees.of(30)), // Position for intaking from ground
+    AMP(Degrees.of(90)), // Position for scoring in amp
+    SPEAKER(Degrees.of(60)); // Position for scoring in speaker
+
+    private final Angle targetAngle;
+    private final Angle angleTolerance;
+
+    ArmPosition(Angle targetAngle, Angle angleTolerance) {
+      this.targetAngle = targetAngle;
+      this.angleTolerance = angleTolerance;
+    }
+
+    ArmPosition(Angle targetAngle) {
+      this(targetAngle, Degrees.of(2)); // 2 degree default tolerance
+    }
+  }
+
+  /**
+   * Gets the current arm position mode.
+   *
+   * @return The current ArmPosition
+   */
+  public ArmPosition getMode() {
+    return currentMode;
+  }
+
+  /**
+   * Sets a new arm position and schedules the corresponding command.
+   *
+   * @param position The desired ArmPosition
+   */
+  public void setArmPosition(ArmPosition position) {
+    currentCommand.cancel();
+    currentMode = position;
+    currentCommand.schedule();
+  }
+
+  // Command that runs the appropriate routine based on the current position
+  private final Command currentCommand =
+      new SelectCommand<>(
+          Map.of(
+              ArmPosition.STOP,
+              Commands.runOnce(this::stop)
+                  .alongWith(Commands.run(() -> checkAtTarget(ArmPosition.STOP)))
+                  .withName("Stop Arm"),
+              ArmPosition.STOWED,
+              createPositionCommand(ArmPosition.STOWED),
+              ArmPosition.INTAKE,
+              createPositionCommand(ArmPosition.INTAKE),
+              ArmPosition.AMP,
+              createPositionCommand(ArmPosition.AMP),
+              ArmPosition.SPEAKER,
+              createPositionCommand(ArmPosition.SPEAKER)),
+          this::getMode);
+
+  /**
+   * Creates a command for a specific arm position that moves the arm and checks the target
+   * position.
+   *
+   * @param position The arm position to create a command for
+   * @return A command that implements the arm movement
+   */
+  private Command createPositionCommand(ArmPosition position) {
+    return Commands.parallel(
+        Commands.runOnce(() -> setPosition(position.targetAngle))
+            .withName("Move to " + position.toString()),
+        Commands.run(() -> checkAtTarget(position))
+            .withName("Check " + position.toString() + " Target"));
+  }
+
+  /**
+   * Checks if the arm is at its target position.
+   *
+   * @return true if at target position, false otherwise
+   */
+  @AutoLogOutput
+  private boolean isAtTarget() {
+    if (currentMode == ArmPosition.STOP) return true;
+    return getPosition().isNear(currentMode.targetAngle, currentMode.angleTolerance);
+  }
+
+  /**
+   * Logs whether the arm is at its target position for a given mode.
+   *
+   * @param position The position to check against
+   */
+  private void checkAtTarget(ArmPosition position) {
```

Comment:
this is a duplicate method as isAtTarget you are just wrapping the logging calls for it. 
so it seems like its extra work and that the logger shoudl log automatically and/or on the periodic?? adding logging here causes it to log a bit more frequently , however, this log call on 233,234 will happen in the operations loop, so this method may be redundant.